api/jsonrpc/server.go

```diff
@@ -235,3 +236,96 @@ func (j *JSONRPCServer) Execute(
 
 	return nil
 }
+
+type SimulateTransactionArgs struct {
+	Tx []byte `json:"tx"`
+}
+
+type SimulatedAction struct {
+	Output []byte  `json:"output_b64"`
+	Error  *string `json:"error"`
+}
+
+type SimulateTransactionReply struct {
+	SimulatedActions []SimulatedAction `json:"simulatedActions"`
+	ReadKeys         []string          `json:"readKeys"`
+	WriteKeys        []string          `json:"writeKeys"`
+	AllocateKeys     []string          `json:"allocKeys"`
+	BlockHeight      uint64            `json:"blockheight"`
+	Error            string            `json:"error"`
+}
+
+type SimulationEnabledRules struct {
+	chain.Rules
+}
+
+func (SimulationEnabledRules) GetTransactionExecutionMode() chain.TransactionExecutionMode {
+	return chain.SimulatedTransactionExecution
+}
+
+func (j *JSONRPCServer) SimulateTransaction(
+	req *http.Request,
+	args *SimulateTransactionArgs,
+	reply *SimulateTransactionReply,
+) error {
+	ctx, span := j.vm.Tracer().Start(req.Context(), "JSONRPCServer.SimulateTransaction")
+	defer span.End()
+
+	if reply == nil {
+		return errors.New("SimulateTransaction was called with a nil reply object")
+	}
```

Comment:
This should be provided by the server, so we can remove this check and follow the style in the rest of the APIs of not checking if the reply value is nil.

Reply:
done